src/core/thread/dua_manager.cpp

```diff
@@ -184,6 +249,522 @@ otError DuaManager::Store(void)
+{
+    Mle::MleRouter &mle = Get<Mle::MleRouter>();
+
+    if (aEvents.Contains(kEventThreadRoleChanged))
+    {
+        if (mle.HasRestored())
+        {
+            UpdateReregistrationDelay();
+        }
+#if OPENTHREAD_CONFIG_DUA_ENABLE && OPENTHREAD_FTD
+        else if (mle.IsRouter())
+        {
+            if (mDelay.mFields.mReregistrationDelay == 0 ||
+                mDelay.mFields.mReregistrationDelay > kNewRouterRegistrationDelay)
+            {
+                // Wait for link establishment with neighboring routers.
+                UpdateRegistrationDelay(kNewRouterRegistrationDelay);
+            }
+        }
+        else if (mle.IsExpectedToBecomeRouter())
+        {
+            // Will check again In case the device decides to stay REED when jitter timeout expires.
+            UpdateRegistrationDelay(mle.GetRouterSelectionJitterTimeout() + 1);
+        }
+#endif
+    }
+
+    if (aEvents.ContainsAny(kEventIp6AddressAdded))
+    {
+        mRegistrationTask.Post();
+    }
+}
+
+void DuaManager::HandleBackboneRouterPrimaryUpdate(BackboneRouter::Leader::State               aState,
+                                                   const BackboneRouter::BackboneRouterConfig &aConfig)
+{
+    OT_UNUSED_VARIABLE(aConfig);
+
+    if (aState == BackboneRouter::Leader::kStateAdded || aState == BackboneRouter::Leader::kStateToTriggerRereg)
+    {
+        UpdateReregistrationDelay();
+    }
+}
+
+void DuaManager::HandleTimer(Timer &aTimer)
+{
+    aTimer.GetOwner<DuaManager>().HandleTimer();
+}
+
+void DuaManager::HandleTimer(void)
+{
+    bool attempt = false;
+
+    otLogDebgDua("regdelay %d, reregdelay %d, checkdelay %d", mDelay.mFields.mRegistrationDelay,
+                 mDelay.mFields.mReregistrationDelay, mDelay.mFields.mCheckDelay);
+
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+    if (mDuaState != kNotExist && TimerMilli::GetNow() > mLastRegistrationTime + Mle::kDuaDadPeriod)
+    {
+        mDomainUnicastAddress.mPreferred = true;
+    }
+
+    if (mDelay.mFields.mRegistrationDelay > 0)
+    {
+        mDelay.mFields.mRegistrationDelay--;
+
+        if (mDelay.mFields.mRegistrationDelay == 0)
+        {
+            attempt = true;
+        }
+    }
+#endif
+
+    if (mDelay.mFields.mCheckDelay > 0)
+    {
+        mDelay.mFields.mCheckDelay--;
+
+        if (mDelay.mFields.mCheckDelay == 0)
+        {
+            attempt = true;
+        }
+    }
+
+    if (mDelay.mFields.mReregistrationDelay > 0)
+    {
+        mDelay.mFields.mReregistrationDelay--;
+
+        if (mDelay.mFields.mReregistrationDelay == 0)
+        {
+#if OPENTHREAD_CONFIG_DUA_ENABLE
+            if (mDuaState != kNotExist)
+            {
+                mDuaState = kToRegister;
+            }
+#endif
+
+#if OPENTHREAD_CONFIG_TMF_PROXY_DUA_ENABLE
+            mChildDuaRegisterMask.Clear();
+#endif
+            attempt = true;
+        }
+    }
+
+    if (attempt)
+    {
+        PerformNextRegistration();
```

Comment:
```suggestion
        mRegistrationTask.Post();
```
What is the rule to determine which to use?

Reply:
Good question. The intention is using task in the CoAP process module to avoid depth stress on stack.
In event handler, previous thought is it is not so time-consuming, so here should be fine to call the function directly.
Now I think it would be better to always use the task  as it varies in children volume and IPAddresses per child in specific project, not sure how long would it take to iterate. Will use task uniformly.

Reply:
Agree that `mRegistrationTask.Post();` is a better way to go.